internal/impl/mysql/aws/aws.go

```diff
@@ -46,26 +46,35 @@ func awsIAMAuth(ctx context.Context, awsConf *service.ParsedConfig, dbConf *mysq
 	}
 
 	var (
-		err    error
-		awsCfg aws.Config
-
+		err         error
+		awsCfg      aws.Config
 		endpoint    string
 		region      string
 		roleConfigs []roleConfig
+
+		opts []func(*awsconfig.LoadOptions) error
 	)
 	if awsCfg, err = awsconfig.LoadDefaultConfig(ctx); err != nil {
 		return nil, fmt.Errorf("unable to load AWS config: %w", err)
 	}
 	if endpoint, err = awsConf.FieldString("endpoint"); err != nil {
 		return nil, err
 	}
-	if region, err = awsConf.FieldString("region"); err != nil {
-		return nil, err
-	} else if region != "" {
-		awsCfg.Region = region
+	if region, _ = awsConf.FieldString("region"); region != "" {
```

Comment:
I'm not a huge fan of errors swallowing.

Reply:
It's fairly harmless on config parse for fields that are optional, just avoids a bit of bloat in the parsing logic.